mempool/v0/clist_mempool.go

```diff
@@ -183,6 +183,16 @@ func (mem *CListMempool) TxsFront() *clist.CElement {
 	return mem.txs.Front()
 }
 
+// GetTxByKey is not supported by the v0 mempool but it is required to satisfy the mempool interface.
+func (mem *CListMempool) GetTxByKey(key types.TxKey) (types.Tx, bool) {
+	return nil, false
+}
+
+// IsTxEvicted is not supported by the v0 mempool but it is required to satisfy the mempool interface.
+func (mem *CListMempool) IsTxEvicted(key types.TxKey) bool {
+	return false
+}
```

Comment:
[blocking] I don't think we have rough consensus on deprecating the v0 mempool. See https://github.com/celestiaorg/celestia-core/issues/1382.

IMO if it's easy to add, then we should add support for `IsTxEvicted` until we make a decision on deprecating the v0 mempool. If it's difficult to add support for `IsTxEvicted` to the v0 mempool, that should be information that we use when making the decision to deprecate or not. If that's the case, making the decision should block merging this PR.

Reply:
if i'm understanding it correctly v0 is currently not in use. I'm not sure why we'd want to support TxStatus in v0 if it's not enabled and won't be used. We could merge this pr as long as we satisfy the interface. 

I have no problem implementing those methods in v0 I just don't really understand why we should

Reply:
The point I think rootul is making is even although it's not the default, it is feasible that someone use the mempool v0, thus we do support it and until we stop supporting it, features like `TxStatus` will need to be implemented across all implementations